### PR TITLE
feature/1666 - Add KCLT

### LINKED ISSUE #1666: Add KCLT - Charlotte-Douglas Int'l Airport
Add CLT to openscope.

### PR DESCRIPTION
**Pull request by MichaelDMcCracken**
Resolves #1666 

The purpose of this pull request is to add KCLT.

### CODE REVIEW

**Comment by MichaelDMcCracken** [comment]
Ready for a review.

File: assets/airports/kclt.json [hunk 1]
```diff
@@ -0,0 +1,7898 @@
...
+                        "KCLT18L": [["_RWY18L1DME", "A12+"], "#183"],
+                        "KCLT18C": [["_RWY18C1DME", "A12+"], "#183"],
+                        "KCLT18R": [["_RWY18R1DME", "A12+"], "#200"],
+                        "KCLT23": ["#235"],
+                        "KCLT36L": [["_RWY36L1DME", "A12+"], "#315"],
+                        "KCLT36C": [["_RWY36C1DME", "A12+"], "#330"],
+                        "KCLT36R": [["_RWY36R1DME", "A12+"], "#025"]
+                    },
+                    "body": ["MESHY", "GULFY", "BARMY", "BATTA"],
+                    "exitPoints": {
```
Comment (Marcel510) [inline assets/airports/kclt.json:381]: L381: BARMY is also a valid exit point (applies to other SIDs as well)

File: assets/airports/kclt.json [hunk 2]
```diff
@@ -0,0 +1,7898 @@
...
+            ],
+            "ils": [true, true]
+        }
+    ],
+    "sids": {
+        "BARMY4": {
+                    "icao": "BARMY4",
+                    "name": "Barmy Four",
+                    "rwy": {
+                        "KCLT05": ["#055"],
```
Comment (Marcel510) [inline assets/airports/kclt.json:371]: L371: "KCLT5" (Even though I'm not sure if that actually makes a difference in the sim, also applies to the other SIDs)

File: assets/airports/kclt.json [hunk 3]
```diff
@@ -0,0 +1,7898 @@
...
+                        ["CLT", "ANDYS", "TREAL*"]
+                    ]
+        },
+         "KILNS4": {
+                    "icao": "KILNS4",
+                    "name": "Kilns Four",
+                    "rwy": {
+                        "KCLT05": ["#055"],
+                        "KCLT18L": [["_RWY18L1DME", "A12+"], "#140"],
+                        "KCLT18C": [["_RWY18C1DME", "A12+"], "#200"],
```
Comment (Marcel510) [inline assets/airports/kclt.json:580]: L579/580: should be #183

File: assets/airports/kclt.json [hunk 4]
```diff
@@ -0,0 +1,7898 @@
...
+                    "name": "Critter Five",
+                    "rwy": {
+                        "KCLT05": ["#055"],
+                        "KCLT18L": [["_RWY18L1DME", "A12+"], "#183"],
+                        "KCLT18C": [["_RWY18C1DME", "A12+"], "#183"],
+                        "KCLT18R": [["_RWY18R1DME", "A12+"], "#200"],
+                        "KCLT23": ["#235"],
+                        "KCLT36L": [["_RWY36L1DME", "A12+"], "#315"],
+                        "KCLT36C": [["_RWY36C1DME", "A12+"], "#330"],
+                        "KCLT36R": [["_RWY36R1DME", "A12+"], "#360"]
```
Comment (Marcel510) [inline assets/airports/kclt.json:639]: Should be heading 025

File: assets/airports/kclt.json [hunk 5]
```diff
@@ -0,0 +1,7898 @@
...
+                    "name": "Queen Four",
+                    "rwy": {
+                        "KCLT05": ["#055"],
+                        "KCLT18L": [["_RWY18L1DME", "A12+"], "#183"],
+                        "KCLT18C": [["_RWY18C1DME", "A12+"], "#183"],
+                        "KCLT18R": [["_RWY18R1DME", "A12+"], "#200"],
+                        "KCLT23": ["#235"],
+                        "KCLT36L": [["_RWY36L1DME", "A12+"], "#315"],
+                        "KCLT36C": [["_RWY36C1DME", "A12+"], "#330"],
+                        "KCLT36R": [["_RWY36R1DME", "A12+"], "#360"]
```
Comment (Marcel510) [inline assets/airports/kclt.json:662]: Should be heading 025

File: assets/airports/kclt.json [hunk 6]
```diff
@@ -0,0 +1,7898 @@
...
+                    "name": "Lills Two",
+                    "rwy": {
+                        "KCLT05": ["#055"],
+                        "KCLT18L": [["_RWY18L1DME", "A12+"], "#183"],
+                        "KCLT18C": [["_RWY18C1DME", "A12+"], "#183"],
+                        "KCLT18R": [["_RWY18R1DME", "A12+"], "#200"],
+                        "KCLT23": ["#235"],
+                        "KCLT36L": [["_RWY36L1DME", "A12+"], "#315"],
+                        "KCLT36C": [["_RWY36C1DME", "A12+"], "#330"],
+                        "KCLT36R": [["_RWY36R1DME", "A12+"], "#360"]
```
Comment (Marcel510) [inline assets/airports/kclt.json:684]: Should be heading 025

File: assets/airports/kclt.json [hunk 7]
```diff
@@ -0,0 +1,7898 @@
...
+                    "name": "Weasel Four",
+                    "rwy": {
+                        "KCLT05": ["#055"],
+                        "KCLT18L": [["_RWY18L1DME", "A12+"], "#183"],
+                        "KCLT18C": [["_RWY18C1DME", "A12+"], "#183"],
+                        "KCLT18R": [["_RWY18R1DME", "A12+"], "#200"],
+                        "KCLT23": ["#235"],
+                        "KCLT36L": [["_RWY36L1DME", "A12+"], "#315"],
+                        "KCLT36C": [["_RWY36C1DME", "A12+"], "#330"],
+                        "KCLT36R": [["_RWY36R1DME", "A12+"], "#360"]
```
Comment (Marcel510) [inline assets/airports/kclt.json:705]: Should be heading 025

File: assets/airports/kclt.json [hunk 8]
```diff
@@ -0,0 +1,7898 @@
...
+    "has_terrain": true,
+    "wind": {
+        "angle": 200,
+        "speed": 8
+    },
+    "arrivalRunway": "18R",
+    "departureRunway": "18C",
+    "defaultMaps": ["Base Map"],
+    "airspace": [
+        {
```
Comment (Marcel510) [inline assets/airports/kclt.json:29]: L29 and below: missing correct indentions

File: documentation/airport-guides/kclt.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
+### About this facility
```
Comment (Marcel510) [inline documentation/airport-guides/kclt.md:1]: The airport guide is quite short, some more information about the airport might be useful. You can also have a look at the other airport guides to get some inspiration.

**Review by Marcel510 — CHANGES_REQUESTED** [review]
@MichaelDMcCracken apologies for the delay. 
Initial review for now, looking good overall, only a couple of minor things.
Please make sure to update the branch with develop since a couple of things have changed since you started working on this (e.g. the airportLoadList.js has been migrated to airportLoadList.json).
Also the restricted areas section is missing, are the no restricted areas around KCLT that we should add to the sim?
As soon as you addressed this review please request another review so that we can go with the review process 👍

File: assets/airports/kclt.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:371]: Fixed

File: assets/airports/kclt.json (hunk 1 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:381]: Fixed

File: assets/airports/kclt.json (hunk 2 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:371]: Fixed

File: assets/airports/kclt.json (hunk 3 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:580]: Fixed

File: assets/airports/kclt.json (hunk 4 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:639]: Fixed

File: assets/airports/kclt.json (hunk 5 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:662]: Fixed

File: assets/airports/kclt.json (hunk 6 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:684]: Fixed

File: assets/airports/kclt.json (hunk 7 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:705]: Fixed

File: assets/airports/kclt.json [hunk 10]
```diff
@@ -0,0 +1,7898 @@
+{
+    "airac": 1813,
```
[the remark this answers is not in the document]
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:2]: Fixed

File: assets/airports/kclt.json (hunk 8 above)
Comment (MichaelDMcCracken) [reply assets/airports/kclt.json:29]: Fixed

File: documentation/airport-guides/kclt.md (hunk 9 above)
Comment (MichaelDMcCracken) [reply documentation/airport-guides/kclt.md:1]: Honestly I don't have a lot of time to put into it. If someone else wants to do this, great, but I can't modify it.

**Comment by MichaelDMcCracken** [comment]
@Fechulo 
Any way to get this approved? Just trying to clear out all my old PRs prior to starting newer stuff. It's literally all done and was reviewed once before.

**Comment by Fechulo** [comment]
@MichaelDMcCracken 
image

**Comment by MichaelDMcCracken** [comment]
> @MichaelDMcCracken image

Fixes pushed. Just for saving my own sanity in the future, how do I pull up THAT for myself ^

**Comment by Fechulo** [comment]
> How do I pull up THAT for myself ^
F12 on Chrome

**Comment by Fechulo** [comment]
@MichaelDMcCracken After playing around with it for a little while, the only thing that could be better is that aircraft on the PARQR3 and CHSLY5 arrivals are coming in too high and always need delay vectors

**Comment by MichaelDMcCracken** [comment]
> @MichaelDMcCracken After playing around with it for a little while, the only thing that could be better is that aircraft on the PARQR3 and CHSLY5 arrivals are coming in too high and always need delay vectors

Yep, it’s just an issue with the planes wanting to stay as high as possible through the published altitude. Probably is something that could be improved on the aircraft model. I’m more than willing to “fake it” to get more realistic altitudes

**Comment by Fechulo** [comment]
You've got my blessing

**Comment by MichaelDMcCracken** [comment]
> You've got my blessing

All good to go now. I moved the window to `A130+|A150-` instead of `A130+|A210-`. Should fix a lot of the problems. I also set JEPHS (CHSLY5.ILS 23) as `A60` (as per our OKC discussion) instead of `A60+`

**Review by Fechulo — APPROVED** [review]

**Comment by MichaelDMcCracken** [comment]
@Fechulo 
Upon further testing, the altitude setting for the CHSLY5 wasn't good enough. I made a few more adjustments and the aircraft are doing better more consistently.

**Pull request merged**